### PR TITLE
Helper class and functions for iFCI fragment import

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
*Highlights*
- Add support for iFCI fragments. Mostly, it is just a manner of supporting `correction=0.` for each term. The rest was renaming the class and adapting the documentation.
- Fixed the import function with the new QEMIST Cloud outputs.
- Tests for iFCI and MIFNO with new QEMIST Cloud outputs.

### CODE REVIEW

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
Small changes, otherwise all good !

Since people know us for MI-FNO and iFCI I think we need to mention the terms at least in the main docstrings of your helper to be crystal clear from the start. By extension this applies to our notebook in your other PR: we should probably rename it mifno_ifci, highlight both acronyms in the intro (opportunity for us to link to something good about IFCI PFAS), and adjust the link in contents.rst

Question: are we then handling both MIFNO and iFCI jobs the same way, without distinction ? From what I've read in this PR one could infer Q.Cloud computed things differently and MIFNO applied a MP2 correction but iFCI did not.

File: tangelo/problem_decomposition/incremental/incremental_helper.py [hunk 1]
```diff
@@ -12,37 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""This file provides helpers in order to import data coming from a MI-FNO job
-run on QEMIST Cloud, providing the users with both fragment information as well as
-reference results obtained by the classical solvers in QEMIST Cloud. The
+"""This file provides helpers in order to import data coming from an incremental
```
[the remark this answers is not in the document]
Comment (AlexandreF-1qbit) [reply tangelo/problem_decomposition/incremental/incremental_helper.py:15]: I added the acronyms and references.

File: tangelo/problem_decomposition/incremental/incremental_helper.py [hunk 2]
```diff
@@ -171,82 +182,73 @@ def frag_info_flattened(self):
...
+                "mo_coeffcients_".
+            suffix (str):  Suffix for the file name structure, including the
+                file extension. Default is ".h5".
         """
-        if not os.path.isdir(destination_folder):
-            raise FileNotFoundError(f"The {destination_folder} path does not exist.")
-        absolute_path = os.path.abspath(destination_folder)
+
+        if not os.path.isdir(source_folder):
+            raise FileNotFoundError(f"The {source_folder} path does not exist.")
```
[the remark this answers is not in the document]
Comment (AlexandreF-1qbit) [reply tangelo/problem_decomposition/incremental/incremental_helper.py]: The error message is not printing the absolute path. When a user is giving an absolute path, the `absolute_path = os.path.abspath(destination_folder)` line just does nothing.

**Comment by AlexandreF-1qbit** [comment]
For your question in the top comment, MIFNO and IFCI are indeed handled in the same way. For iFCI, there is no correction term. The MO coefficient are different, as iFCI has a better way of handling orbitals and the truncation scheme is "exact" within the approximations they are working with.

File: tangelo/problem_decomposition/incremental/incremental_helper.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ValentinS4t1qbit) [reply tangelo/problem_decomposition/incremental/incremental_helper.py]: Ok :)

My suggestions remain the same, here

**Comment by AlexandreF-1qbit** [comment]
I added two tests for the `retrieve_mo_coeff` method. The `compute_fermionoperator` is not included in the test, as the underlying mechanisms of computer a fermionic operator is already tested in `SecondQuantizedMoleculeTests`.

**Review by ValentinS4t1qbit — APPROVED** [review]
Thank you Alexandre ! We'll figure out the qiskit issues separately, I see no reason to delay this.

**Pull request merged**